pkgs/top-level/python-packages.nix

```diff
@@ -7185,9 +7185,22 @@ in {
 
   samsungtvws = callPackage ../development/python-modules/samsungtvws { };
 
+  sanic = callPackage ../development/python-modules/sanic {
+    # Disable tests to prevent loops, as they require Sanic itself.
+    sanic-routing = self.sanic-routing.overridePythonAttrs (oldAttrs: {
+      doCheck = false;
+    });
+    sanic-testing = self.sanic-testing.overridePythonAttrs (oldAttrs: {
+      doCheck = false;
+      dontUsePythonImportsCheck = true;
+    });
+  };
```

Comment:
This is the wrong way round: We should only override packages that are not in `propagatedBuildInputs`.

Reply:
Any suggestions? Off the top of my head, tests have to be disabled either here or at the top level, or else we have to build everything twice (first with tests off, then use those to build with tests on).

Reply:
`sanic-routing` and `sanic-testing` have `sanic` in their `checkInputs`, so we can override their input `sanic`.

Reply:
If I got this right, it'd the last option I proposed: 1) build `sanic-routing` and `sanic-testing` without tests; 2) use those as input to an intermediate `sanic`; 3) use it to build `sanic-routing` and `sanic-testing` with tests; and 4) use those as inputs to the final `sanic`.

And the lines you highlighted go into the overrides for `sanic-routing` and `sanic-testing`, leaving the top-level `sanic` with no overrides.

Reply:
Oh, true. I forgot that `sanic` doesn't build without those dependencies. If there's no way to do that we should just turn off `sanic-routing`'s and `sanic-testing`'s tests.

Reply:
So it turns out that `sanic-routing` doesn't actually depend on `sanic` for anything, despite it being listed in `tests_require`, so that package is no longer an issue.

As for `sanic-testing`, my reading of the situation is:
- `sanic` only depends on `sanic-testing` for tests and some backwards compatibility (see also https://github.com/sanic-org/sanic/pull/2009);
- `sanic-testing` does depend on `sanic` at runtime (the imports check can't even be done when it's not present).

Therefore I've added `sanic` as a propagated build input of `sanic-testing`, with special logic to disable all testing when the former is explicitly set to `null` (which we do when building `sanic`). Said special logic is only to avoid having to call both `override` and `overridePythonAttrs` in sequence in `python-packages.nix`.

All in all, this should give us three fully tested packages at the top level.